src/main/java/service/impl/LegendFilterImpl.java

```diff
@@ -0,0 +1,22 @@
+package service.impl;
+
+import dao.InputDao;
+import dao.impl.InputDaoImpl;
+import java.util.List;
+import service.LegendFilter;
+
+public class LegendFilterImpl implements LegendFilter {
+    private static final String LEGEND_REGEX = "\\w++,\\w++,\\w++";
+    private static final String CORRECT_DATA_REGEX = "[bspr],\\w++,\\d++";
+    private static final int LEGEND_INDEX = 0;
+    private final InputDao storageService = new InputDaoImpl();
+
+    @Override
+    public void getLegendFiltered() {
+        List<String> inputData = storageService.getStorageData();
+        String legend = storageService.getStorageData().get(LEGEND_INDEX);
+        if (legend.matches(LEGEND_REGEX) && !legend.matches(CORRECT_DATA_REGEX)) {
+            inputData.remove(LEGEND_INDEX);
+        }
+    }
```

Comment:
Looks like you are just filtering out the invalid lines, but we should not skip any parts of the data that the user sent to us this is inconsistent behavior and we should not make decisions for the user on how to handle that. 
In most cases, if we encounter some error - we throw an exception and cancel the whole transaction if anything went wrong, this is called 'atomic operations' went from the databases and 'ACID', you can google that if you want, and we will learn that in the future

Overall lets modify it and move to the `TransactionParser`, to check the format of the lines

Reply:
okay(